pkg/build/controller/policy/policy.go

```diff
@@ -11,6 +11,7 @@ import (
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
 	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildlister "github.com/openshift/origin/pkg/build/generated/listers/build/internalversion"
```

Comment:
Maybe I'm saying the wrong thing ... but you're creating a buildLister in line 102 in build_controller.go. As far as I can tell, that's using a cache. 
```
+	buildLister := params.BuildInformer.Lister()
```

that's fine to use for build pruning, which we do use here https://github.com/smarterclayton/origin/blob/76f000014cf26e022b3b05c00bfe8b082429e291/pkg/build/controller/build/build_controller.go#L617

I think you need to create a separate one like:
```
policyBuildLister := buildclient.NewOSClientBuildLister(params.OpenshiftClient)
```
to pass to `policy.GetAllRunPolicies(...)`